pkg/types/validation/installconfig.go

```diff
@@ -566,8 +566,13 @@ func validateNamedRepository(r string) error {
 		// in an error. Instead we will check whether the input is
 		// a valid hostname as a workaround.
 		if err == dockerref.ErrNameNotCanonical {
-			_, err := url.ParseRequestURI(r)
+			// If the hostname string contains a port, lets attempt
+			// to split them
+			host, _, err := net.SplitHostPort(r)
 			if err != nil {
+				host = r
+			}
+			if err = validate.Host(host); err != nil {
 				return fmt.Errorf("the repository provided is invalid")
```

Comment:
Let's wrap around the error returned from `validate.Host`.
```suggestion
				return errors.Wrap(err, "the repository provided is invalid")
```

Reply:
Usually I like wrapping things but this one is unnecessarily verbose. This is what gets spit out:
```Invalid value: "ocp/release-x.y": the repository provided is invalid: a lowercase RFC 1123 subdomain must consist of lower case alphanumeric characters, '-' or '.', and must start and end with an alphanumeric character (e.g. 'example.com', regex used for validation is '[a-z0-9]([-a-z0-9]*[a-z0-9])?(\.[a-z0-9]([-a-z0-9]*[a-z0-9])?)*')
```

Reply:
I personally prefer the more verbose error rather than a statement that the repository is invalid without any details about how it is invalid.

Reply:
Done.
Just so that you know, I was cursing at you while escaping the regex in the validation. :smile:

Reply:
Fair enough. For what it's worth, I would have happily accepted `.*` for the majority of the error message.